### PR TITLE
MULE-9225: If the @XML annotation is not present, the extension should not fail

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java [hunk 1]
```diff
@@ -29,30 +30,52 @@
 public final class XmlModelEnricher extends AbstractAnnotatedModelEnricher
 {
 
-    public static final String DEFAULT_SCHEMA_LOCATION_MASK = "http://www.mulesoft.org/schema/mule/extension/%s";
+    public static final String DEFAULT_SCHEMA_LOCATION_MASK = "http://www.mulesoft.org/schema/mule/%s";
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java]: why public?

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java [hunk 2]
```diff
@@ -29,30 +30,52 @@
...
         Xml xml = extractAnnotation(describingContext.getDeclarationDescriptor().getDeclaration(), Xml.class);
+        DeclarationDescriptor descriptor = describingContext.getDeclarationDescriptor();
+        descriptor.withModelProperty(XmlModelProperty.KEY, createXmlModelProperty(xml, descriptor));
+    }
+
+    private XmlModelProperty createXmlModelProperty(Xml xml, DeclarationDescriptor descriptor)
+    {
+        Declaration declaration = descriptor.getDeclaration();
+        String schemaVersion = declaration.getVersion();
+        String namespace = buildDefaultNamespace(declaration.getName());
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java]: don't calculate stuff until you know you need it

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java [hunk 3]
```diff
@@ -41,11 +43,12 @@
     @Test
     public void enrich()
     {
+        declarationDescriptor.named(EXTENSION_NAME).onVersion(EXTENSION_VERSION);
         XmlModelProperty xmlProperty = enrich(XmlSupport.class);
 
         assertThat(xmlProperty, is(notNullValue()));
         assertThat(xmlProperty.getSchemaVersion(), is(SCHEMA_VERSION));
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: why is schema version different from the extension version in this case? per your code, the schema version defaults to the extension version

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java [hunk 4]
```diff
@@ -41,11 +43,12 @@
...
-        assertThat(xmlProperty.getNamespace(), is(NAMESPACE));
+        assertThat(xmlProperty.getNamespace(), is(NAMESPACE.toLowerCase()));
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: don't do toLowerCase() everywhere, just have the constant to be lowercase

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java (hunk 3 above)
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: Ok I got it, but rename the test to enrichWithCustomSchemaVersion or something like that

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java [hunk 5]
```diff
@@ -53,14 +56,41 @@ public void enrich()
...
+    public void enrichWithCustomNamespaceValue()
+    {
+        declarationDescriptor.named(EXTENSION_NAME).onVersion(EXTENSION_VERSION);
         XmlModelProperty xmlProperty = enrich(DefaultXmlExtension.class);
 
         assertThat(xmlProperty, is(notNullValue()));
         assertThat(xmlProperty.getSchemaVersion(), is(EXTENSION_VERSION));
-        assertThat(xmlProperty.getNamespace(), is(NAMESPACE));
-        assertThat(xmlProperty.getSchemaLocation(), equalTo(String.format(XmlModelEnricher.DEFAULT_SCHEMA_LOCATION_MASK, EXTENSION_NAME)));
+        assertThat(xmlProperty.getNamespace(), is(NAMESPACE.toLowerCase()));
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: same as before

File: modules/extensions-support/src/test/java/org/mule/module/extension/HeisenbergExtension.java [hunk 6]
```diff
@@ -33,21 +31,16 @@
 
 import javax.inject.Inject;
 
-@Extension(name = HeisenbergExtension.EXTENSION_NAME, description = HeisenbergExtension.EXTENSION_DESCRIPTION)
+@Extension(name = HeisenbergExtension.HEISENBERG, description = HeisenbergExtension.EXTENSION_DESCRIPTION)
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/module/extension/HeisenbergExtension.java]: Why does this matter? The end result should be the same

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java (hunk 1 above)
Comment (juandesi) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java]: mistake

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java (hunk 3 above)
Comment (juandesi) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: makes sense

Comment (juandesi) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: This is just information to fill the declarationDescriptor in order to test, if no Name/Version are specified both will be null values that is a case that is never going to happen when building an extension.

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java [hunk 7]
```diff
@@ -29,30 +31,47 @@
...
-            DeclarationDescriptor descriptor = describingContext.getDeclarationDescriptor();
-            descriptor.withModelProperty(XmlModelProperty.KEY, createXmlModelProperty(xml, descriptor));
-        }
+        DeclarationDescriptor descriptor = describingContext.getDeclarationDescriptor();
+        descriptor.withModelProperty(XmlModelProperty.KEY, createXmlModelProperty(xml, descriptor));
     }
 
     private XmlModelProperty createXmlModelProperty(Xml xml, DeclarationDescriptor descriptor)
     {
+        boolean xmlIsPresent = xml != null;
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java]: i'll let it pass but convention is usually isXmlPresent

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java [hunk 8]
```diff
@@ -29,30 +31,47 @@
...
+        namespace = StringUtils.isBlank(namespace) ? declarationName : namespace;
+        return NameUtils.hyphenize(namespace);
+
     }
+
+    private String removeFromName(String name, String word)
+    {
+        return StringUtils.removeEndIgnoreCase(name, word);
+    }
+
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricher.java]: remove empty lines

File: modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java [hunk 9]
```diff
@@ -53,14 +57,41 @@ public void enrich()
...
         declarationDescriptor.named(EXTENSION_NAME).onVersion(EXTENSION_VERSION);
+        XmlModelProperty xmlProperty = enrich(NoXmlSupport.class);
+
+        assertThat(xmlProperty, is(notNullValue()));
+        assertThat(xmlProperty.getSchemaVersion(), is(EXTENSION_VERSION));
+        assertThat(xmlProperty.getNamespace(), is(EXTENSION_HYPHENAZED_NAME));
+        assertThat(xmlProperty.getSchemaLocation(), equalTo(String.format(DEFAULT_SCHEMA_LOCATION_MASK, EXTENSION_HYPHENAZED_NAME)));
+    }
+
+
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/capability/xml/XmlModelEnricherTestCase.java]: remove one empty line

**Pull request merged**